### PR TITLE
Noise module

### PR DESCRIPTION
**Pull request by the-moliver**
I moved dropout to a new noise.py module and added multiplicative Gaussian noise (from the Dropout paper) as well. According to the paper this "Gaussian dropout" actually works slightly better, and it seems to in the MNIST example. I updated all the docs and examples to comply with the changes.

### CODE REVIEW

**Comment by fchollet** [comment]
Not a big fan of breaking everybody's code by removing the Dropout layer from the core module (Dropout has got to be used in almost every Keras model ever). 

There is no particular semantic unity to the layers in the core module, basically these are commonly used layers that are not specific to a particular type of model (recurrent/convolutional/etc). I still think that's where Dropout belong.

Otherwise, we can definitely add Gaussian noise in a noise module. Do you have any numbers showing the benefits of Gaussian noise on an example (compared to Dropout, or at least compared to no regularization)?

**Comment by madisonmay** [comment]
For what it's worth, it's also simple enough to alias the Dropout layer in core and move the logic to noise if so desired.

Regardless, would be neat to see the addition of the Gaussian noise module.  It's described in the [original Dropout paper](http://www.cs.toronto.edu/~rsalakhu/papers/srivastava14a.pdf) in section 10.  According to the paper's authors, it should be "equivalent" to standard dropout under the proper conditions. 

> In this paper, we described dropout as a method where we retain units with probability p
at training time and scale down the weights by multiplying them by a factor of p at test time.
Another way to achieve the same effect is to scale up the retained activations by multiplying
by 1/p at training time and not modifying the weights at test time. These methods are
equivalent with appropriate scaling of the learning rate and weight initializations at each
layer.

@the-moliver curious to see your benchmarks anyhow, though!
